qiskit/transpiler/passes/synthesis/high_level_synthesis.py

```diff
@@ -156,16 +167,75 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
 
 
 class DefaultSynthesisClifford(HighLevelSynthesisPlugin):
-    """The default clifford synthesis plugin."""
+    """The default clifford synthesis plugin.
+
+    For N <= 3 qubits this is the optimal CX cost decomposition by Bravyi, Maslov.
+    For N > 3 qubits this is done using the general non-optimal greedy compilation
+    routine from reference by Bravyi, Hu, Maslov, Shaydulin.
+    """
 
     def run(self, high_level_object, **options):
         """Run synthesis for the given Clifford."""
         decomposition = synth_clifford_full(high_level_object)
         return decomposition
 
 
-class DefaultSynthesisLinearFunction(HighLevelSynthesisPlugin):
-    """The default linear function synthesis plugin."""
+class AGSynthesisClifford(HighLevelSynthesisPlugin):
+    """Clifford synthesis plugin based on the Aaronson-Gottesman method."""
```

Comment:
It'd be good to say explicitly in the docstring here (and for the other plugin classes) that the name of this plugin is `"ag"` so that users know how to specify the plugin.